### PR TITLE
executor: set the correct handle in DirtyDB when executing update statements

### PR DESCRIPTION
**Pull request by zimulala**
## What have you changed? (mandatory)

Set the correct handle in DirtyDB when executing update statements

## What is the type of the changes? (mandatory)
- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)

UT

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.
## Does this PR affect tidb-ansible update? (mandatory)

No.
## Does this PR need to be added to the release notes? (mandatory)

No.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
Please add an integration test.

File: executor/write_test.go [hunk 1]
```diff
@@ -1713,6 +1713,24 @@ func (s *testSuite) TestUpdateSelect(c *C) {
...
+	tk.MustExec("CREATE TABLE ttt (id bigint(20) NOT NULL, host varchar(30) NOT NULL, PRIMARY KEY (id), UNIQUE KEY i_host (host));")
+	tk.MustExec("insert into ttt values (7,7),(8,8),(9,9);")
+
+	tk.MustExec("begin")
+	tk.MustExec("update ttt set id = 0, host='7' where id = 7 limit 1;")
+	tk.MustExec("delete from ttt where id = 0 limit 1;")
+	tk.MustExec("update ttt set id = 0, host='9' where id = 9 limit 1;")
+	tk.MustExec("delete from ttt where id = 0 limit 1;")
+	tk.MustExec("update ttt set id = 0, host='8' where id = 8 limit 1;")
+	tk.MustExec("delete from ttt where id = 0 limit 1;")
```
Comment (jackysp) [inline executor/write_test.go]: It's better to add a row check here, like
`tk.MustQuery("select * from ttt use index (i_host) order by id;").Check(testkit.Rows())`.

**Comment by shenli** [comment]
@zimulala Please fix the CI.

**Pull request merged**